padertorch/data/fragmenter.py

```diff
@@ -107,3 +117,147 @@ def fragment(key, x):
             fragment = deflatten(fragment)
             fragments.append(fragment)
         return fragments
+
+
+def get_segment_boundaries(num_samples, length, shift=None,
+                           mode='begin', choose_one=False, rng=np.random):
+    """
+
+    Args:
+        num_samples: num samples of signal for which boundaries are caclulated
+        length: segment length
+        shift: shift between segments
+        mode: Defines the position of the boundaries in the signal:
+            begin: boundaries start at sample zero
+                so that only values at the end are cut
+            end: boundaries end at num_samples
+                so that only values at the beginning are cut
+            center: boundaries are chosen such that the same number of samples
+                are discarded at the end and the beginning of the signal
+            random: Starts the first segment randomly at a value
+                between 0 and shift. This may reduce the number of segments.
+            random_max_segments: Randomly chooses the first segment such that
+                the maximum number of segments are created
+        choose_one: If True only outputs one segment boundary
+        rng: random number generator (numpy.random)
+
+    Returns:
+        2xB numpy array with start and end values for B boundaries
+
+    >>> np.random.seed(0)
+    >>> get_segment_boundaries(25, 10, None, mode='begin')
+    array([[ 0, 10],
+           [10, 20]])
+    >>> get_segment_boundaries(25, 10, 3, mode='begin')
+    array([[ 0,  3,  6,  9, 12, 15],
+           [10, 13, 16, 19, 22, 25]])
+    >>> get_segment_boundaries(24, 10, 3, mode='begin')
+    array([[ 0,  3,  6,  9, 12],
+           [10, 13, 16, 19, 22]])
+    >>> get_segment_boundaries(24, 10, 3, mode='random')
+    array([[ 0,  3,  6,  9, 12],
+           [10, 13, 16, 19, 22]])
+    >>> get_segment_boundaries(24, 10, 3, mode='random_max_segments')
+    array([[ 1,  4,  7, 10, 13],
+           [11, 14, 17, 20, 23]])
+    >>> get_segment_boundaries(24, 10, 3, mode='center')
+    array([[ 1,  4,  7, 10, 13],
+           [11, 14, 17, 20, 23]])
+    >>> get_segment_boundaries(24, 10, 3, mode='center', choose_one=True)
+    array([[ 1],
+           [11]])
+    """
+    assert num_samples >= length, (num_samples, length)
+    if shift is None:
+        shift = length
+    assert shift > 0, shift
+
+    if mode == 'begin':
```

Comment:
I think the terms are still not very clear yet. Can we come up with a better name for mode? Maybe boundary_mode? Also begin (btw I think begin is a verb while beginning is the corresponding noun) is ambiguous as it may refer to the beginning of the segments (according to Christoph's comment) or to the beginning of the input. Maybe it is better to call it boundary_mode="flush-left" and  boundary_mode="flush-right" instead of "begin" and "end", respectively. What do the others think? @boeddeker @thequilo

Reply:
You are right, "begin" and "end" are verbs and it can be confusing if we use these. Same for start/stop: I don't know what I would expect of the boundary mode "stop". I like the naming with "flush", but I'm not so sure about left and right. I'm fine with them, but is it always clear what left and right mean in this context?

Reply:
I agree, flush is more obvious. Is `flush-begin` something resonable?